core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -955,11 +955,11 @@ func (o *evmTxStore) SaveFetchedReceipts(ctx context.Context, r []*evmtypes.Rece
 // NOTE: We continue to attempt to resend evm.txes in this state on
 // every head to guard against the extremely rare scenario of nonce gap due to
 // reorg that excludes the transaction (from another wallet) that had this
-// nonce (until finality depth is reached, after which we make the explicit
+// nonce (until LatestFinalizedBlockNum is reached, after which we make the explicit
 // decision to give up). This is done in the EthResender.
 //
 // We will continue to try to fetch a receipt for these attempts until all
-// attempts are below the finality depth from current head.
+// attempts are below the LatestFinalizedBlockNum from current head.
```

Comment:
nit: think the new logic is inclusive of the latest finalized block num so this comment needs to be updated. Something like
```suggestion
// attempts are equal to or below the LatestFinalizedBlockNum from current head.
```